src/Directory.Build.props

```diff
@@ -14,4 +14,12 @@
 
   </PropertyGroup>
 
+  <ItemGroup>
```

Comment:
You may want add a condition like `Condition="'$(UseMLCodeAnalyzer)' != 'false'"` so that projects could opt out.  Supporting such a condition requires this to be in `Directory.Build.targets` so that it is imported after the project body where a project might set this property.

Reply:
> Supporting such a condition requires this to be in Directory.Build.targets so that it is imported after the project body where a project might set this property.

All MSBuild properties are evaluated first, then Items are evaluated in a second pass.  So it doesn't necessarily need to be in a `.targets` file.

Reply:
Actually this was something I was curious about with how MSBuild works, and I couldn't find the documentation on this... so let me see if I have this right @eerhardt  and @ericstj ... so imported projects (including implicitly imported projs like `Directory.Build.props`, their *properties* (that is, I suppose, things in `PropertyGroup` tags) are evaluated before the local importing project file, but their items (things in `ItemGroup` tags) are evaluated after the local importing project file? Is that so?

Reply:
> all MSBuild properties are evaluated first, then Items are evaluated in a second pass.

Right you are.  Still for sanity's sake its usually better to follow logical order.

@TomFinley here are the docs: https://msdn.microsoft.com/en-us/library/dd997067.aspx#Anchor_2.  Like @eerhardt said you don't necessarily need to move it when adding the condition, but I like to do that by convention for sanity's sake.  In general I try to use props files only when I need to define something that a project might need to consume itself or I need it defined before some other targets are imported.  Also if you want a visual representation of the import resolution you can run /pp on the project file (/pp:project.pp).

Reply:
Ah OK... so *all* properties are evaluated in the order in which they occur no matter where they may be, then *all* items, I guess? And the `Directory.Build.props` (if found) is always implicitly imported first? That begins to make some sense, I think.

Reply:
Directory.Build.props (and targets) are just imported by convention in the SDK props/targets, which get imported before/after the project body respectively.  /pp output shows you precisely where (which is sometimes relevant when trying to override).